src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -723,62 +725,64 @@ private Model GetVocabularyDictionary()
-                                        model.AddWordVector(ch, key, value);
-                                    else
-                                        ch.Warning($"Parsing error while reading model file: '{_modelFileNameWithPath}', line number {lineNumber + 1}");
-                                }
-                            }
-                            lineNumber++;
-                        }
+                            (bool isSuccess, string key, float[] values) = LineParser.ParseKeyThenNumbers(line);
+
+                            if (isSuccess)
+                                parsedData.Add((key, values, lineNumber + skippedLinesCount));
+                            else // we use shared state here (ch) but it's not our hot path and we don't care about unhappy-path performance
+                                ch.Warning($"Parsing error while reading model file: '{_modelFileNameWithPath}', line number {lineNumber + skippedLinesCount}");
+                        });
 
-                        // Handle first line of the embedding file separately since some embedding files including fastText have a single-line header
-                        string[] wordsInFirstLine = firstLine.TrimEnd().Split(delimiters);
-                        dimension = wordsInFirstLine.Length - 1;
+                    Model model = null;
+                    foreach (var parsedLine in parsedData.OrderBy(parsedLine => parsedLine.lineNumber))
+                    {
+                        dimension = parsedLine.values.Length;
                         if (model == null)
                             model = new Model(dimension);
-                        if (model.Dimension == dimension)
-                        {
-                            float temp;
-                            string firstKey = wordsInFirstLine[0];
-                            float[] firstValue = wordsInFirstLine.Skip(1).Select(x => float.TryParse(x, out temp) ? temp : Single.NaN).ToArray();
-                            if (!firstValue.Contains(Single.NaN))
-                                model.AddWordVector(ch, firstKey, firstValue);
-                        }
-                        pch.Checkpoint(lineNumber);
+                        if (model.Dimension != dimension)
+                            ch.Warning($"Dimension mismatch while reading model file: '{_modelFileNameWithPath}', line number {parsedLine.lineNumber}, expected dimension = {model.Dimension}, received dimension = {dimension}");
+                        else
+                            model.AddWordVector(ch, parsedLine.key, parsedLine.values);
                     }
+
+                    // Handle first line of the embedding file separately since some embedding files including fastText have a single-line header
+                    var firstLine = File.ReadLines(_modelFileNameWithPath).First();
+                    string[] wordsInFirstLine = firstLine.TrimEnd().Split(' ', '\t');
```

Comment:
Why doesn't this call `LineParser.ParseKeyThenNumbers`?

Reply:
because `ParseKeyThenNumbers` expects that every line is sth like `string float float float float ...`. If the first line of some file would be `string string string` the current code would return dimension = 2, the `ParseKeyThenNumbers` would return 0. It's a single line so there would be no noticeable perf gain and I did not want to risk.